### ISSUE TITLE
Multi-paragraph lists move first paragraph to new line with `list-style-position: inside`

### ISSUE THREAD

**Issue by pranabekka** [raphlinus/pulldown-cmark#735]
If I have a list with extra paragraphs or other elements, it wraps the first line inside a `<p>` element, which moves the paragraph to a line by itself when I style the list with `list-style-position-inside`. The second list (item) in the image shows the issue.

![list-position-inside-example](https://github.com/raphlinus/pulldown-cmark/assets/20282735/710a1aaa-5117-4499-9930-45793ea95a82)
The first list (item) shows the what the desired output would look like, and the second list (item) shows what the actual/current output would look like.

**Input**:
```
- Hello world

  This list has multiple "blocks" of text
```

**Actual Output** (second list in image):
```
<ul>
  <li>
    <p>Hello world</p>
    <p>This list has two paragraph ("p") blocks</p>
  </li>
</ul>
```

**Desired Output** (first list in image):
```
<ul>
  <li>
    Hello world
    <p>This list has two "blocks" of text</p>
  </li>
</ul>
```

**Description:**
Basically, the first "paragraph" shouldn't (ideally/for me) be wrapped in a `<p>` tag, because if the first item in a list item is a block element, it gets put on a new line (given `list-style-position: inside`. I just figured out a workaround with `li > p:nth-child(1) { display: inline }`, so it's not a big deal, but I thought I'd share it, since it's all written down anyway.

**Version:** 0.9
Might be 0.9.1 — I'm using [Zola v0.17.2](https://github.com/getzola/zola/releases/tag/v0.17.2), which was released in March.
